graphql_compiler/cost_estimation/statistics.py

```diff
@@ -0,0 +1,211 @@
+        return self._class_count[class_name]
+
+    def get_edge_count_between_vertex_pair(
+        self, vertex_out_class_name, vertex_in_class_name, edge_class_name
+    ):
+        """Return number of edges of given class connecting two vertex types.
+
+        For cost estimation, sometimes we need to predict traversal costs between two vertex types
+        using an edge type that connects the vertices' abstract types. If this statistic isn't
+        provided, it can be roughly estimated using multiple calls of get_class_count().
+
+        Args:
+            vertex_out_class_name: str, vertex class name.
+            vertex_in_class_name: str, vertex class name.
+            edge_class_name: str, edge class name.
+
+        Returns:
+            - int, the count of edges if the statistic exists
+            - None otherwise
+        """
+        statistic_id = (vertex_in_class_name, vertex_out_class_name, edge_class_name)
+        if statistic_id not in self._edge_count_between_vertex_pair:
+            return None
+        return self._edge_count_between_vertex_pair[statistic_id]
+
+    def get_domain_count(self, vertex_name, field_name):
+        """Return the number of distinct values the vertex field has over all vertex instances.
+
+        This statistic helps estimate the result size of @filter directives used to restrict
+        values using equality operators like '=', '!=', and 'in_collection'.
+
+        Preconditions:
+        1. field_name must be a valid vertex field of vertex_name
+
+        Args:
+            vertex_name: str, name of a vertex.
+            field_name: str, name of a vertex field.
+
+        Returns:
+            - int, domain size of the vertex field's values if the statistic exists
+            - None otherwise
+        """
+        statistic_id = (vertex_name, field_name)
+        if statistic_id not in self._domain_count:
+            return None
+        return self._domain_count[statistic_id]
+
+    def get_histogram(self, vertex_name, field_name):
+        """Return a histogram for the given vertex field, providing statistics about range values.
+
+        This statistic helps estimate the result size of @filter directives used to restrict
+        numbers using inequality operators like '>', '<', and 'between'.
+
+        Preconditions:
+        1. field_name must be a valid vertex field of vertex_name
+
+        Args:
+            vertex_name: str, name of a vertex.
+            field_name: str, name of a vertex field.
+
+        Returns:
+            - list[tuple(int, int, int)], histogram as a list of tuples with 3 elements each if the
+              statistic exists. Each tuple represents a bin as (start, end, elementCount), where
+              [start, end) is the range of values in the bin, and elementCount is the bin element
+              count.
+            - None otherwise
+        """
+        if vertex_name not in self._histogram or field_name not in self._histogram[vertex_name]:
```

Comment:
This could just as easily be `self._histogram.get(vertex_name, dict()).get(field_name, None)`

Reply:
To be honest, that reads like something I'd see in a code-golf. I much prefer the version we have now. 
If I'm missing something or this is something much better, please tell me.

Reply:
Let's ignore for a second the performance cost of those key lookups (there are 5 in yours and 2 in mine).

The remaining difference is this: in my one-liner, we only specify *once* what key goes first and what key goes second. In yours, count in how many places it is either implicitly or explicitly defined that `vertex_name` is the first key, and `field_name` the second? Each one of those places (after the first) is an opportunity to make a mistake and introduce a bug.

Reply:
I think that the best solution is the following:
```
histogram_key = (vertex_name, field_name)
`return self._histograms.get(histogram_key)
```
It is consistent with the style of the other methods and avoids the complexity of having a nested dict.(Note that I renamed `self._histogram` to `self._histograms`. Please see comment above.

Reply:
I'll push this in another PR shortly.